app/controllers/papers_controller.rb

```diff
@@ -33,6 +33,8 @@ def __quote(val)
   end
 
   def search
+    redirect_to login_path unless signed_in?
```

Comment:
Wondering if I should add a `notice: "Please sign in."` to this.

Reply:
i think it doesn't hurt to add it, if someone happens to be visiting the url directly :)